### ISSUE TITLE
Updated ttf hinted binaries in repo

### ISSUE THREAD

**Issue by IceMan81** [google/roboto#152]
Will you be updating the hinted binaries checked into repo?

**Comment by jamesgk** [comment]
I don't think that will happen any time soon, because no one is actively hinting updated binaries.

We can update the unhinted binaries, though.

**Comment by IceMan81** [comment]
Are they manually hinted or using some tool like ttfautohint?

**Comment by jamesgk** [comment]
Traditionally they have been manually hinted.

**Comment by IceMan81** [comment]
OK. Having the updated unhinted binaries would still be useful. Thanks.

**Comment by davelab6** [comment]
On 18 February 2016 at 20:05, Gesly George <notifications@github.com> wrote:

> OK. Having the updated unhinted binaries would still be useful. Thanks.


Would it be even more useful for them to be ttfautohinted

**Comment by IceMan81** [comment]
Yes & No. I was planing on running ttfautohint agaist the unhinted binaries myself, that way I can control the ttfautohint parameters and use the latest freetype libs (patched with infinality). But, if someone just wanted to get the latest version fonts for use hassle free, having ttfautohinted binaries would be useful.

**Comment by roozbehp** [comment]
I believe we should remove the unhinted binaries but put them in releases instead https://help.github.com/articles/about-releases/

As for autohinted binaries, we would appreciate parameters from people with experience in autohinting. We can put the parameters in the tree and autohinted binaries in the releases too.

**Comment by IceMan81** [comment]
These are the parameters I use when I use ttfautohint based on their documentation & recommendations I've come across.

`-l 2 -r 200 -G 200 -W -p -c -I -t -w G`

**Comment by jamesgk** [comment]
OK, I created a release for a recent, fairly stable build of the TTFs: https://github.com/google/roboto/releases/tag/v2.129. I'll submit a PR to remove the TTFs from version control.

**Comment by IceMan81** [comment]
Are these the hinted or the unhinted version?

**Comment by jamesgk** [comment]
Unhinted. I'll keep the hinted versions in the repository (they're not overwritten by running the tools).

**Comment by davelab6** [comment]
Would you like more information about ttfautohint?

**Comment by codeman38** [comment]
On the release page, could you add a ZIP file containing all of the TTFs? Currently it looks like the only way to download them is one at a time, which is awkward.

**Comment by davelab6** [comment]
If so, please include the license file with the TTFs :)

**Comment by jamesgk** [comment]
That's done, though I noticed just recently that https://github.com/google/roboto/releases/latest (which points to v2.129 as of this posting) contains incorrect small caps mappings as described at https://github.com/behdad/fonttools/issues/504#issuecomment-195013791.

What's troubling is that the tests don't catch this; I'll update them and create a new release with this fixed ASAP (luckily this issue did not seem to propogate to other releases e.g. Android).

**Comment by jamesgk** [comment]
Latest release fixed; keeping this open for the task of including hinted binaries in releases

**Comment by IceMan81** [comment]
@jamesgk The hinted version in the just released v2.132 looks like old versions that don't match the version of the unhinted fonts.

**Comment by jamesgk** [comment]
In some ways, yes. Basically it's versions of the fonts in hinted/ which have had some relatively easy fixes applied to them (specifically, those in [touchup_for_web.py](https://github.com/google/roboto/blob/master/scripts/touchup_for_web.py)). So we still don't have autohinted binaries in the release (thus this issue is still open).

**Comment by adrientetar** [comment]
So there are no plans to have hints source in the repository built with the fonts?

**Comment by jamesgk** [comment]
I guess the goal is to include autohinted fonts built from source in the releases. But there is no specific plan to do so.

**Comment by jamesgk** [comment]
Since a ttfautohint wrapper was added to fontmake (https://github.com/googlei18n/fontmake/pull/104), I've been experimenting with this a bit. @davelab6 (or anyone else that's interested), would you like to take a look at the attached autohinted Roboto output? "roboto_ah1.ttf" was created without any arguments to ttfautohint, "roboto_ah2.ttf" was created with arguments `-l 2 -r 200 -G 200 -W -p -c -I -t -w G`. "roboto_h.ttf" is the baseline font found in the web release.

Personally, I didn't find any difference in the outputs with or without arguments. There was a difference between manually and autohinted fonts but it didn't seem significant to me.

[roboto.zip](https://github.com/google/roboto/files/362431/roboto.zip)

**Comment by davelab6** [comment]
Awesome! Im happy to take a look

**Comment by jamesgk** [comment]
I pushed branch https://github.com/google/roboto/tree/ttfautohint with the code to produce these fonts, if that's helpful.
